packages/guui/components/Dropdown/index.js

```diff
@@ -193,18 +199,21 @@ export default class Dropdown extends Component<
         const checkboxID = `checkbox-id-${this.props.id}`;
 
         return (
-            <Div>
+            <div>
                 {this.state.noJS ? (
                     [
                         <label
                             htmlFor={checkboxID}
-                            className={this.state.isExpanded ? 'expanded' : ''}
+                            className={
```

Comment:
We've discussed elsewhere such as the[ incoming style rules](https://github.com/guardian/dotcom-rendering/commit/6b57464) that we should be using `cx` to conditionally render classNames. This `button` like a good use case for  using cx to conditionally apply class names here. For example...

```js
const button = css`
    color: blue;
    font-weight: 700;
`;

const expanded = css`
    color: red;
`;

const Button = ({ isExpanded }) =>
    <button className={cx({
        button: true,
        expanded: isExpanded
    )}}>Click Me</a>;
```

Reply:
I just tried this in my PR and AFAICT it'll need to be

```
const Button = ({ isExpanded }) =>
    <button className={cx({
        [button]: true,
        [expanded]: isExpanded
    )}}>Click Me</a>;
```

 or

 ```
const Button = ({ isExpanded }) =>
    <button className={cx(
        button,{
        [expanded]: isExpanded
    )}}>Click Me</a>;
```
I prefer the second form. I think the benefit is more in terms of standardising how and where logic in applying styles should be applied and belong. I think at this level of complexity, ternarys would be fine- but they'd start to lose out with any greater complexity.

Reply:
Hey, thanks and completely agree that cx pays off as things get more complex. It just feels like what you've written is, for the current use case at least - which is quite a common one, a lot less readable than what's currently there.

Reply:
As I understand it, there's a performance reason for using `cx` instead of the ternary. 

With the ternary, styles for both `button` and `buttonExpanded` would always be shipped, regardless of whether `isExpanded` is true or false. `buttonExpanded` is essentially all the styles in `button` duplicated, plus a few extras.

With `cx`, only styles for `button` and `buttonExpanded` (which in our case would contain no duplication, just the styles `:after` pseudo-selector) would be shipped if is `isExpanded` was false.

If this is difficult to visualise, you can try playing around with this [codesandbox](https://codesandbox.io/s/w2z5mzoxol). But as a rule of thumb, I think it's worth assuming that `cx` _always tries to optimise your CSS output_, whereas ternaries do not.

Reply:
Ah that's helpful. It makes sense. I guess that in this case there is the change of having to generate styles at runtime/client-side but that feels okay.